python_modules/libraries/dagster-dbt/dagster_dbt/dbt_artifacts.py

```diff
@@ -0,0 +1,164 @@
+                Default: "target"
+            prepare_command: The dbt cli command to run to prepare the manifest.json
+                Default: ["parse", "--quiet"]
+            packaged_project_dir (Union[str, Path]):
+                A directory that will contain a copy of the dbt project and the manifest.json
+                when the artifacts have been built. The prepare method will handle syncing
+                the project_dir to this directory.
+                This is useful when the dbt project needs to be part of the python package data
+                like when deploying using PEX.
+        """
+        self.project_dir = Path(project_dir)
+        if not self.project_dir.exists():
+            raise DagsterDbtProjectNotFoundError(f"project_dir {project_dir} does not exist.")
+        self._target_folder = Path(target_folder)
+        self._prepare_command = prepare_command
+        # this is ok if it doesn't exist, will get created by `prepare`
+        self._packaged_project_dir = Path(packaged_project_dir) if packaged_project_dir else None
+
+        if self._should_use_built_artifacts():
+            self._manifest_path = self._current_project_dir.joinpath(
+                self._target_folder, "manifest.json"
+            )
+        else:
+            self._manifest_path = self._prepare_manifest()
+
+    @public
+    @property
+    def manifest_path(self) -> Path:
+        """The path to the manifest.json, compiling the manifest first if in development or
+        ensuring it already exists if not.
+        """
+        if not self._manifest_path.exists():
+            if self._should_use_built_artifacts():
+                raise DagsterDbtManifestNotPreparedError(
+                    f"Did not find prepared manifest.json at expected path {self._manifest_path}.\n"
+                    "If this is an environment that is expected to prepare the manifest at run time, "
+                    "set the environment variable DAGSTER_DBT_PARSE_PROJECT_ON_LOAD."
+                )
+            else:
+                raise DagsterDbtManifestNotPreparedError(
+                    f"Unexpected state, {self._manifest_path} should have been created when this "
+                    "object was initialized."
+                )
+
+        return self._manifest_path
+
+    @public
+    def prepare(self, *, quiet=False) -> None:
+        """A method that can be called as part of the deployment process to handle
+        preparing the manifest and if packaged_project_dir is set, handle copying
+        the dbt project to that directory.
+
+        Args:
+            quiet (bool):
+                Disable logging
+                Default: False
+        """
+        level = logging.DEBUG if quiet else logging.INFO
+        self._prepare_manifest(level)
+        self._handle_package_data(level)
+        logger.log(level, "Preparation complete.")
+
+    @public
+    def get_cli_resource(self, **kwargs) -> DbtCliResource:
```

Comment:
Would like to see how this is used in an example before introducing it. 

Are we expecting the following? Feels weird to force the user to instantiate `DbtCliResource` outside of its class.

```python
defs = Definitions(
    ...,
    resources={
        "dbt": my_artifacts.get_cli_resource(...)
    },
)
```

Would prefer something else like the following, so `DbtCliResource` is explicit, and we continue to have typing on `DbtCliResource` instantiation.

```python
defs = Definitions(
    ...,
    resources={
        "dbt": DbtCliResource(artifacts=my_artifacts, **kwargs)
    },
)
```

Reply:
> Would like to see how this is used in an example before introducing it.

https://github.com/dagster-io/dagster/pull/19925/files is what ive been working with 

Agree with your take, will pull for now and put up a PR for `artifacts` on `DbtCliResource`